ios/Gekidou/Sources/Gekidou/Networking/ShareExtension.swift

```diff
@@ -36,18 +41,27 @@ struct UploadSessionData {
         let fileIds = dict["fileIds"] as! [String]
         let message = dict["message"] as! String
         let totalFiles = dict["totalFiles"] as! Int
+        let failNotificationID = dict["failNotificationID"] as! String
+        let failTimeout = dict["failTimeout"] as! Date
         
         return UploadSessionData(
             serverUrl: serverUrl,
             channelId: channelId,
             files: files,
             fileIds: fileIds,
             message: message,
-            totalFiles: totalFiles
+            totalFiles: totalFiles,
+            failNotificationID: failNotificationID,
+            failTimeout: failTimeout
         )
     }
 }
 
+// TODO: setup some configuration for allowing users to figure out the right ammount of time to wait.
+let SHARE_TIMEOUT: Double = 120 // change to 300
+let NETWORK_ERROR_MESSAGE: String = "Mattermost App couldn't acknowledge the message was posted due to network conditions, please review and try again"
+let FILE_ERROR_MESSAGE: String = "The shared file couldn't be processed for sharing"
```

Comment:
Ideally this would be localized. You can check the code in the Share Extension around the Views on how is done with the use of NSLocalizedString or String.localizedString.